storage/aptosdb/src/transaction_store/mod.rs

```diff
@@ -177,6 +177,45 @@ impl TransactionStore {
         })
     }
 
+    /// Get write sets in `[begin_version, end_version)` half-open range.
+    ///
+    /// N.b. an empty `Vec` is returned when `begin_version == end_version`
+    pub fn get_write_sets(
+        &self,
+        begin_version: Version,
+        end_version: Version,
+    ) -> Result<Vec<WriteSet>> {
```

Comment:
I am wondering if it's better to return an iterator for this API, just like the `get_transaction_iter` API? That way, we can avoid allocating a large amount of memory if this `WriteSet` range is too big?

Reply:
I think at this point it's an over optimization. 😆